### PR TITLE
Simplify drop_intents_below_freq

### PR DESCRIPTION
**Pull request by Mazyod**
**Proposed changes**:
`drop_intents_below_freq` suffers from a bug where not all `TrainingData` content is being copied over to the newer object. The proposal fixes this issue and simplifies the code.

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by Mazyod** [comment]
I am currently cloning Rasa repo to run black and see if I can add some unit tests, however, the cloning process is extremely slow.

File: rasa/nlu/test.py [hunk 1]
```diff
@@ -166,17 +166,12 @@ def drop_intents_below_freq(
...
-    return TrainingData(
-        keep_examples,
-        training_data.entity_synonyms,
-        training_data.regex_features,
-        responses=training_data.responses,
+    examples_per_intent = training_data.number_of_examples_per_intent.items()
+    intents_below_cutoff = {i for i, c in examples_per_intent if c < cutoff}
+
+    return training_data.filter_training_examples(
+        lambda ex: ex.get(INTENT) not in intents_below_cutoff
```
Comment (Mazyod) [inline rasa/nlu/test.py:174]: Alternatively...

```suggestion
    return training_data.filter_training_examples(
        lambda ex: examples_per_intent[ex.get(INTENT)] >= cutoff
```

**Comment by sara-tagger** [comment]
Thanks for submitting a pull request 🚀 @wochinge will take a look at it as soon as possible ✨

**Review by Ghostvv — COMMENTED** [review]
great change! the fix looks good. Could you please add tests where old code fails, but the new one works

**Comment by Mazyod** [comment]
@Ghostvv I have update the existing test such that it fails on main, and succeeds on my patch. Not sure if I should've included the lookups in the json file or just inject them as I did.

**Review by Ghostvv — APPROVED** [review]
looks good to me

**Pull request merged**